### PR TITLE
Refactor Proposals Page

### LINKED ISSUE #329: Refactor Proposals page
Refactor proposals page: 
- Abstract Table and Filter as separate components (index, styled, types, ...). 
- Abstract utils and helpers in the page to be part of hooks. 
- define useProposals hook, with a way to manage proposals, filter (filter, setFilter) and see how we can manage filter logic separated from the components and still share it.
- write minimal tests if possible.

**Blocked by:**
- #328 
- #307 
- #340

### PR DESCRIPTION
**Pull request by hhamud**
* Closes #329

### CODE REVIEW

File: src/components/Proposals/Search/SchemaSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  const {onFilterChange,  getParams} = useParams('scheme', 'All Schemes')
+
+
+  useEffect(() => {
+    setSchemeFilter(getParams)
+    dispatch({
+      type: 'filter',
+      payload: {
+        searc: state.filters.search,
```
Comment (rossneilson) [inline src/components/Proposals/Search/SchemaSearch.tsx]: Typo in seach

File: src/components/Proposals/Search/SchemaSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  cursor: pointer;
+  width: 200px;
+  padding: 0px 10px;
+  margin: 10px 0px;
+  font-family: var(--roboto);
+  border: 0px;
+  align-self: center;
+`;
+
+const SchemaSearch = () => {
```
Comment (rossneilson) [inline src/components/Proposals/Search/SchemaSearch.tsx]: I think we should stick to calling them schemes. I may be wrong but don't think schema and scheme are interchangeable?
Maybe `SchemeSearch` would be better?

File: tsconfig.json [hunk 3]
```diff
@@ -23,14 +23,14 @@
...
-    "noUnusedParameters": true,
+    "noUnusedParameters": false,
     "noImplicitReturns": true,
     "allowUnreachableCode": false,
     "isolatedModules": true,
     "useDefineForClassFields": true,
     "noFallthroughCasesInSwitch": true,
     "useUnknownInCatchVariables": false,
-    "noUnusedLocals": true,
+    "noUnusedLocals": false,
```
Comment (rossneilson) [inline tsconfig.json]: Why are we disabling this?

**Review by rossneilson — COMMENTED** [review]
Just a few small comments, otherwise the refactor looks great

File: tsconfig.json (hunk 3 above)
Comment (hhamud) [reply tsconfig.json]: I forgot to turn this back on

**Review by rossneilson — APPROVED** [review]

**Pull request merged**